src/System.Net.Http.WinHttpHandler/tests/UnitTests/WinHttpHandlerTest.cs

```diff
@@ -846,6 +846,20 @@ public void SendAsync_NonSecureRequestToServer_ExpectNoClientCertContext()
                 async () => { await client.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, cts.Token); });
         }
 
+        [Fact]
+        public async Task SendAsync_WinHttpOpenReturnsError_ExpectHttpRequestException()
+        {
+            var handler = new WinHttpHandler();
+            var client = new HttpClient(handler);
+            var request = new HttpRequestMessage(HttpMethod.Get, FakeServerEndpoint);
+
+            TestControl.Fail.WinHttpOpen = true;
+
+            Exception ex = await Assert.ThrowsAsync<HttpRequestException>(
+                async () => { await client.SendAsync(request); });
```

Comment:
> But interestingly, if I add surrounding braces to my one line statement, it doesn't compile

Yes, that's expected.  ThrowsAsync is expecting a ```Func<Task>```.  When you give it ```() => client.SendAsync(request)```, you're giving it a task-returning delegate because SendAsync returns a Task.  But when you give it ```() => { client.SendAsync(request); }```, you're giving it a void-returning delegate.